tests/keras/preprocessing/test_image.py

```diff
@@ -160,6 +160,33 @@ def test_directory_iterator(self):
         assert(sorted(dir_iterator.filenames) == sorted(filenames))
         shutil.rmtree(tmp_folder)
 
+    ''' Test for the directory iterator when we wish to use it for a FCN.'''
+    def test_directory_iterator_fcn(self):
```

Comment:
I suggest also checking if labels can be loaded from pngs without palette conversion as described in the following, perhaps an additonal color mode 'segmentation_label' would be appropriate:
https://github.com/nicolov/segmentation_keras/issues/14

Though perhaps something like that is out of scope for this pull request.